cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -21,25 +21,32 @@ fn low8willSXto32(x: u32) -> bool {
 // "enc" in the following means "hardware register encoding number".
 
 #[inline(always)]
-fn mkModRegRM(m0d: u8, encRegG: u8, rmE: u8) -> u8 {
+fn encode_modrm(m0d: u8, enc_reg_g: u8, rm_e: u8) -> u8 {
     debug_assert!(m0d < 4);
-    debug_assert!(encRegG < 8);
-    debug_assert!(rmE < 8);
-    ((m0d & 3) << 6) | ((encRegG & 7) << 3) | (rmE & 7)
+    debug_assert!(enc_reg_g < 8);
+    debug_assert!(rm_e < 8);
+    ((m0d & 3) << 6) | ((enc_reg_g & 7) << 3) | (rm_e & 7)
 }
 
 #[inline(always)]
-fn mkSIB(shift: u8, encIndex: u8, encBase: u8) -> u8 {
+fn encode_sib(shift: u8, enc_index: u8, enc_base: u8) -> u8 {
     debug_assert!(shift < 4);
-    debug_assert!(encIndex < 8);
-    debug_assert!(encBase < 8);
-    ((shift & 3) << 6) | ((encIndex & 7) << 3) | (encBase & 7)
+    debug_assert!(enc_index < 8);
+    debug_assert!(enc_base < 8);
+    ((shift & 3) << 6) | ((enc_index & 7) << 3) | (enc_base & 7)
 }
 
-/// Get the encoding number from something which we sincerely hope is a real
-/// register of class I64.
+/// Get the encoding number of a GPR.
 #[inline(always)]
-fn iregEnc(reg: Reg) -> u8 {
+fn int_reg_enc(reg: Reg) -> u8 {
+    debug_assert!(reg.is_real());
+    debug_assert_eq!(reg.get_class(), RegClass::I64);
+    reg.get_hw_encoding()
+}
+
+/// Get the encoding number of any register.
+#[inline(always)]
+fn reg_enc(reg: Reg) -> u8 {
```

Comment:
What is this used for?  I don't see a previous version of it, and I feel nervous about having a function that gets the encoding for a register without checking the class.  (cf inconsistency with `int_reg_enc`).  At some point soon I guess we'll need an `xmm_reg_enc` as a dual to `int_reg_enc`?

Reply:
There were two TODOs (removed in this patch) that said that two `int_reg_enc` calls should really be replaced with a new `reg_enc`, because registers that could flow in could be either GPRs or SSE. Hence this new function.